packages/twenty-server/src/engine/metadata-modules/flat-field-metadata/validators/utils/validate-enum-flat-field-metadata.util.ts

```diff
@@ -18,128 +18,172 @@ import { FieldMetadataExceptionCode } from 'src/engine/metadata-modules/field-me
 import { type FlatFieldMetadataValidationError } from 'src/engine/metadata-modules/flat-field-metadata/types/flat-field-metadata-validation-error.type';
 import { type FlatFieldMetadata } from 'src/engine/metadata-modules/flat-field-metadata/types/flat-field-metadata.type';
-import { runFlatFieldMetadataValidators } from 'src/engine/metadata-modules/flat-field-metadata/utils/run-flat-field-metadata-validators.util';
-import { type FlatMetadataValidator } from 'src/engine/metadata-modules/types/flat-metadata-validator.type';
-import {
-  beneathDatabaseIdentifierMinimumLength,
-  exceedsDatabaseIdentifierMaximumLength,
-} from 'src/engine/metadata-modules/utils/validate-database-identifier-length.utils';
+import { IDENTIFIER_MAX_CHAR_LENGTH } from 'src/engine/metadata-modules/utils/constants/identifier-max-char-length.constants';
+import { IDENTIFIER_MIN_CHAR_LENGTH } from 'src/engine/metadata-modules/utils/constants/identifier-min-char-length.constants';
 import { isSnakeCaseString } from 'src/utils/is-snake-case-string';
 
 const validateMetadataOptionId = (sanitizedId?: string) => {
-  const validators: FlatMetadataValidator<string>[] = [
-    {
-      validator: (id) => !isDefined(id),
-      message: msg`Option id is required`,
-    },
-    {
-      validator: (id) => !z.string().uuid().safeParse(id).success,
-      message: msg`Option id is invalid`,
-    },
-  ];
-
-  return runFlatFieldMetadataValidators({
-    elementToValidate: sanitizedId,
-    validators,
-  });
+  const errors: FlatFieldMetadataValidationError[] = [];
+
+  // Check if id is defined
+  if (!isDefined(sanitizedId)) {
+    errors.push({
+      code: FieldMetadataExceptionCode.INVALID_FIELD_INPUT,
+      message: t`Option id is required`,
+      userFriendlyMessage: msg`Option id is required`,
+    });
+  }
+
+  // Check if id is a valid UUID
+  if (!z.string().uuid().safeParse(sanitizedId).success) {
+    errors.push({
+      code: FieldMetadataExceptionCode.INVALID_FIELD_INPUT,
+      message: t`Option id is invalid`,
+      userFriendlyMessage: msg`Option id is invalid`,
+      value: sanitizedId,
+    });
+  }
+
+  return errors;
 };
 
 const validateMetadataOptionLabel = (
   sanitizedLabel: string,
 ): FlatFieldMetadataValidationError[] => {
+  const errors: FlatFieldMetadataValidationError[] = [];
+
+  // Check if label is defined
   if (!isDefined(sanitizedLabel)) {
-    return [
-      {
-        code: FieldMetadataExceptionCode.INVALID_FIELD_INPUT,
-        message: t`Option label is required`,
-        userFriendlyMessage: msg`Option label is required`,
-      },
-    ];
+    errors.push({
+      code: FieldMetadataExceptionCode.INVALID_FIELD_INPUT,
+      message: t`Option label is required`,
+      userFriendlyMessage: msg`Option label is required`,
+    });
+
+    return errors;
   }
 
+  // Check if label is a non-empty string
```

Comment:
Could rm comment will do in next PR if not other feedback requiring ci re-trigger